checker/env.go

```diff
@@ -99,6 +101,9 @@ func NewEnv(container *containers.Container, provider ref.TypeProvider, opts ...
 	if envOptions.crossTypeNumericComparisons {
 		filteredOverloadIDs = make(map[string]struct{})
 	}
+	if envOptions.validatedDeclarations != nil {
+		declarations = envOptions.validatedDeclarations.Copy()
```

Comment:
It's not possible with protos to somehow have a "copy on write" type here is it?  Being able to share the declarations across multiple environments would be a potentially use memory optimization here...

Reply:
In this case only a shallow copy is made of the map containing the protos. The actual protos themselves for function declarations are `proto.Clone`d if there's an overload being added to a function that's already in the declaration set.